src/Node/Stream.purs

```diff
@@ -74,19 +90,55 @@ onData r cb =
       Right buf ->
         pure buf
 
+read :: forall w eff. Readable w (err :: EXCEPTION | eff) -> Eff (err :: EXCEPTION | eff) (Maybe Buffer)
+read r = do
+  v <- read' r
+  case v of
+       Nothing -> return Nothing
```

Comment:
Can you please use `pure` instead of `return`? `return` is being removed in the next Prelude release.